### PR TITLE
Actix 4 and Image Optimization

### PR DESCRIPTION
**Pull request by SaltyAom**
## TLDR;
- Better performance.
- Move to Actix Web 4.
- Docker image size reduced to 1.92MB
- Add compile flag and image optimization.

Enjoy the benefit~

Happy doog

## More details
Confused doog

This PR introduced, moving from Actix web 3 to Actix Web 4 beta.
Although Actix 4 is in beta, from investigation and testing, Actix 4 performed well and has no bug or any issues with nhql.

### Actix 4
Actix 4 moved its base using Tokio 1 is the main reason I decided to move to 4 beta, it's worth a try since most of Rust package also is moving to Tokio 1 as soon as possible as it's considered as mature and has stable API.

Overall, I see a significantly lesser package being installed (which reduced bundle size) and slightly better performance by moving to Actix 4 alone.

I also refactor image type mapping and removed `lazy_static` as it's not quite a good idea to install a package just to map an extension, instead it's refactored to use simple `match` instead.

### Docker image
Previously, nhql use Debian Distroless but as nhql is such a tiny stateless micro-service, I think most people don't really tap into image of small stateless image really.
So I moved nhql to use Scratch image as it doesn't have any runtime exception thanks to Rust.

I also add some optimization to the image while building and image thus increasing performance and reducing image size.
For instance, using upx, adding Rust flag to native-cpu, building for specific platform (Linux musl x86_64).

The total image size is reduced to 1.92 MB.

However, using scratch, Rust native TLS in Linux doesn't work, which will talk in the next section.

### Panic = abort
Some Rust flag is also added but what's worth the note is `panic = 'abort'`. 

As nhql is really small and stateless, every action has already been planned and predicted.

It's really hard to find a spot to exploit this situation as it would be almost impossible to panic as I carefully crafted it up.
However, if it's really the case, most image is run in some orchestration which normally would have some logging which can track why the node is crushing.

As long as I created and deployed [nhql in production](https://api.opener.studio/graphiql) on Google Cloud Run, it have never been panic yet.
For this PR is already been deployed for quite some weeks, which is actually replaced the old production and it has never been panic yet even with the new production of [Atago Opener](https://atago.opener.studio) which has traffic quite almost all the time.

And still, it never panic.

## Rust TLS
By default, Reqwest (Rust XHR library) uses native TLS which Distroless has some supply support.
However, native TLS of Linux environments like Alpine and Scratch is not supported in this case which causes an unexpected runtime error.

So this PR fixes the issue by adding Rust TLS instead.
This should not require any change with your SSL certificate.

## Conclusion
That's it.
Update dependencies, optimize image, add some flag and it's faster and better!

Happy doog

### CODE REVIEW

**Comment by SaltyAom** [comment]
LGTM, merged.

confirmed
